packages/web/src/components/tipping/support/TopSupporters.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { MouseEvent, useCallback } from 'react'
+
+import { IconTrophy, IconArrow } from '@audius/stems'
+import cn from 'classnames'
+import { useDispatch } from 'react-redux'
+
+import { useSelector } from 'common/hooks/useSelector'
+import { getProfileUser } from 'common/store/pages/profile/selectors'
+import { getSupporters } from 'common/store/tipping/selectors'
+import { UserProfilePictureList } from 'components/notification/Notifications/UserProfilePictureList'
+import {
+  setUsers,
+  setVisibility
+} from 'store/application/ui/userListModal/slice'
+import {
+  UserListEntityType,
+  UserListType
+} from 'store/application/ui/userListModal/types'
+
+import styles from './Support.module.css'
+
+const MAX_TOP_SUPPORTERS = 5
+
+const messages = {
+  topSupporters: 'Top Supporters',
+  viewAll: 'View All'
+}
+
+export const TopSupporters = () => {
+  const dispatch = useDispatch()
+  const profile = useSelector(getProfileUser)
+  const supportersMap = useSelector(getSupporters)
+  const supportersForProfile = profile
+    ? supportersMap[profile.user_id] ?? {}
+    : {}
+  const rankedSupportersList = Object.keys(supportersForProfile)
+    .sort((k1, k2) => {
+      const id1 = parseInt(k1)
+      const id2 = parseInt(k2)
+      return supportersForProfile[id1].rank - supportersForProfile[id2].rank
+    })
+    .map(k => supportersForProfile[parseInt(k)])
+
+  const handleClick = useCallback(
+    (event: MouseEvent<HTMLDivElement>) => {
+      if (profile) {
+        const target = event.target as HTMLDivElement
+        const className = (target?.className ?? '').toString()
+        const isProfilePicture = className.includes(
```

Comment:
not in love with checking target for this! if we are going to, maybe it makes sense to use a `data-testid` or other attr that is less subject to change than classname? down to chat about a pattern here! other option is maybe just passing onClick down to the profile pictures?

Reply:
Do we even need to pass it down? Would changing `UserProfilePictureList`'s default `onClick` handler for the images to call `event.stopPropagation()` work here?